### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by Therealadetunji**
Create a method called current_user in your ApplicationController to make current user data available to all controllers.
It will return the first user from the database. ✔️ 
Don't worry about proper authentication at this time. You will add it in the upcoming projects.
Create forms to perform the following functions: ✔️ 
Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
Allow Users to add likes to Posts. ✔️

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1 +1,28 @@
...
+<h3>post #<%= post.id %></h3>
+<h4><%= post.title %></h4>
+<h5><%= post.text %></h5>
+
+<h6>Comments: <%= post.commentscounter %> Likes: <%= post.likescounter %></h6>
+</div>
+ <% end %>
+
+ <button class="btn">Pagination</button>
+
```
Comment (OLIPLICHE) [inline app/views/posts/index.html.erb:28]: - Please make sure that there is a form in the `app/views/posts/new.html.erb` file which accepts input for creating posts.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    @comment.save if @comment.valid?
+    redirect_to user_post_path(current_user, @post.id)
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)[:text]
+  end
+end
```
Comment (OLIPLICHE) [inline app/controllers/comments_controller.rb:23]: - Please make sure there should be a pair of routes in the `config/routes.rb` for creating comments
- **Hint**: When you run `rails routes`, you have to get this: `POST /posts/:post_id/comments(.:format) comments#creat` and `GET /posts/:post_id/comments/new(.:format) comments#new`  like what I paste in post controller
- 
**NB: This comment was supposed to be lined with `Route` file, unfortunetly you did not make the file visible**

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+      else
+        render :new, status: 'Error: Post was not created.'
+      end
+    end
   end
 end
+
+def post_param
+  params.require(:post).permit(:title, :text)
+end
```
Comment (OLIPLICHE) [inline app/controllers/posts_controller.rb:31]: - Job well done in `config/route` file 🥇 
- 
Capture

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = current_user.comments.new(
+      author_id: current_user.id,
+      post_id: @post.id,
+      text: comment_params
+    )
+
```
Comment (OLIPLICHE) [inline app/controllers/comments_controller.rb:13]: - Please there should be a form in the `views/comments/new.html.erb` file which accepts input for creating comments.


**NB: This comment was supposed to be lined with `views/comments/new.html.erb` file, unfortunetly you did not make the file visible**

File: app/controllers/likes_controller.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+      post_id: @post.id
+    )
+
+    if @like.save
+      redirect_to user_post_path(current_user, @post)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+end
```
Comment (OLIPLICHE) [inline app/controllers/likes_controller.rb:16]: - Please there should be a route in the `config/routes.rb` file for creating like

**NB: This comment was supposed to be lined with `Route` file, unfortunately, you did not make the file visible**

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Therealadetunji, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
Linter is green (the last commit have a green check mark (✅)) 👏
README file is descriptive 👏
Commit messages are meaningful 👏 👏
Gitflow is correctly used (Pull Request is made from the feature branch into the dev branch) 👏

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @Therealadetunji 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**